lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3584,7 +3593,8 @@ void SPIRVToLLVM::transIntelFPGADecorations(SPIRVValue *BV, Value *V) {
       // Alloca might be hidden by a couple of casts.
       bool isStaticMemoryAttribute = AL ? true : false;
       while (!isStaticMemoryAttribute && Inst &&
-             (isa<BitCastInst>(Inst) || isa<AddrSpaceCastInst>(Inst))) {
+             (isa<BitCastInst>(Inst) || isa<AddrSpaceCastInst>(Inst) ||
+              isa<GetElementPtrInst>(Inst))) {
```

Comment:
When we have a(n FPGA) attribute attached to structure members - GEP is being annotated. So stripping it is incorrect (that's why here we don't use `stripPointerCasts`, since it also strips GEP(0,0) and you never know if it's access to the first member of a structure or not).